p2p/server.py

```diff
@@ -233,24 +237,37 @@ def _make_syncer(self, peer_pool: PeerPool) -> BaseService:
         return FullNodeSyncer(
             self.chain, self.chaindb, self.base_db, peer_pool, self.cancel_token)
 
-    def _make_peer_pool(self, discovery: DiscoveryProtocol) -> PeerPool:
+    def _make_peer_pool(self, fut_discovery: 'asyncio.Future[DiscoveryProtocol]') -> PeerPool:
         # This method exists only so that ShardSyncer can provide a different implementation.
         return self.peer_pool_class(
             self.peer_class,
             self.headerdb,
             self.network_id,
             self.privkey,
-            discovery,
+            fut_discovery,
             max_peers=self.max_peers,
         )
 
-    async def _run(self) -> None:
-        self.logger.info("Running server...")
-        upnp_dev = await self._discover_upnp_device()
+    async def _make_discovery(self,
+                              upnp_device: Awaitable[upnpclient.Device]) -> DiscoveryProtocol:
```

Comment:
Passing in a single upnp device will not be sufficient for discovery. Take a look at the proposed refactor here, which loops over multiple devices: https://github.com/ethereum/py-evm/pull/905

Reply:
Yep, that commit basically just tries to solve the problem of not having the `PeerPool` available with the instantiating of the `Server`. It's a small refactor which doesn't change the rest of the process. I can rebase that in case #905 lands first.